service/src/test/java/bio/terra/workspace/pact/WdsContractVerificationTest.java

```diff
@@ -78,44 +118,183 @@ void pactVerificationTestTemplate(PactVerificationContext context) {
+            "Expected authenticated request for %s, got %s", expectedEmail, request.getEmail()));
   }
 
-  @State({"a workspace with the given id exists"})
-  public void workspaceIdExists(Map<?, ?> parameters) {
-    var workspaceUuid = UUID.fromString(parameters.get("id").toString());
+  @State("a workspace with the given {id} exists")
+  public void workspaceIdExists(Map<?, ?> pactState) {
+    var workspaceUuid = getUuid(pactState, "id");
     when(mockWorkspaceDao.getWorkspace(eq(workspaceUuid)))
         .thenReturn(
             Workspace.builder()
                 .workspaceId(workspaceUuid)
-                .workspaceStage(WorkspaceStage.RAWLS_WORKSPACE)
-                .userFacingId("workspaceid")
+                .workspaceStage(WorkspaceStage.MC_WORKSPACE)
+                .userFacingId("userfacingid")
                 .createdByEmail("workspace.creator@e.mail")
                 .state(WsmResourceState.READY)
                 .build());
   }
 
-  @State({"policies allowing snapshot reference creation"})
+  @State("an Azure cloud context exists for the given {workspace_id}")
+  public void workspaceIsAzure(Map<?, ?> pactState) {
+    var workspaceUuid = getUuid(pactState, "workspace_id");
+    when(mockWorkspaceDao.getCloudContext(workspaceUuid, CloudPlatform.AZURE))
+        .thenReturn(
+            Optional.of(
+                new DbCloudContext()
+                    .cloudPlatform(CloudPlatform.AZURE)
+                    .state(WsmResourceState.READY)));
+  }
+
+  @State({
+    "no Azure storage container resources exist for the given {workspace_id}",
+    "no snapshot resources exist for the given {workspace_id}"
+  })
+  public void noResourcesExist(Map<?, ?> pactState) {
+    var workspaceUuid = getUuid(pactState, "workspace_id");
+    when(mockResourceDao.enumerateResources(
+            eq(workspaceUuid),
+            any(WsmResourceFamily.class),
+            any(StewardshipType.class),
+            /* offset= */ anyInt(),
+            /* limit= */ anyInt()))
+        .thenReturn(Collections.emptyList());
+  }
+
+  @State("an Azure storage container resource exists for the given {workspace_id}")
+  public Map<String, String> storageContainerExists(Map<?, ?> pactState) {
+    var workspaceUuid = getUuid(pactState, "workspace_id");
+    String storageContainerName = String.format("sc-%s", workspaceUuid);
+    ControlledAzureStorageContainerResource stubbedStorageContainerResource =
+        ControlledAzureStorageContainerResource.builder()
+            .storageContainerName(storageContainerName)
+            .common(
+                ControlledResourceFields.builder()
+                    .name(storageContainerName)
+                    .workspaceUuid(workspaceUuid)
+                    .createdByEmail("storage.container.creator@e.mail")
+                    // TODO(for reviewer): are these reasonable defaults for storage
+                    //   container?
+                    .managedBy(ManagedByType.MANAGED_BY_USER)
```

Comment:
Looking for input on this...I tried to guess at what the most relevant values were, and I'm not sure they _really_ matter but if I can make more reasonable assumptions I'd like to.

Reply:
Does this match what's returned by WSM for a real actual resource? If so, seems fine

Reply:
I was going off of what was in Swagger & what the validation errors were telling me...good call to just look at a prod resource...adjusted (trivially) in a357b1196e7aaa416646eacd0aca579ba4eecf8b